doc/source/ray-core/scheduling/ray-oom-prevention.rst

```diff
@@ -0,0 +1,234 @@
+
+    /tmp/ray/session_latest/logs/python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_60002.log:[2022-10-12 16:14:18,843 I 60002 60031] task_manager.cc:458: task c8ef45ccd0112571ffffffffffffffffffffffff01000000 retries left: 3, oom retries left: 0, task failed due to oom: 1
+
+    /tmp/ray/session_latest/logs/python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_60002.log:[2022-10-12 16:14:18,843 I 60002 60031] task_manager.cc:466: No retries left for task c8ef45ccd0112571ffffffffffffffffffffffff01000000, not going to resubmit.
+
+.. note::
+
+    Task retries are executed immediately. If there is a long running process it is possible for a task to keep retrying and fail and exhaust the oom retries. Consider increasing ``RAY_task_oom_retries``, or :ref:`limit the number of concurrently running tasks <core-patterns-limit-running-tasks>`.
+
+.. note::
+
+    Actors by default are non-retriable since :ref:`max_restarts <actor-fault-tolerance>` defaults to 0, therefore tasks are preferred to actors when it comes to what gets killed first. Actor right now doesn't use ``RAY_task_oom_retries`` and instead uses :ref:`max_restarts <actor-fault-tolerance>` when it is killed by the memory monitor. This is to be changed very soon.
+
+Worker killing policy
+~~~~~~~~~~~~~~~~~~~~~
+
+The raylet prioritizes killing tasks that are retriable, i.e. when ``max_retries`` or ``max_restarts`` is > 0. This is done to minimize workload failure. It then looks for the last one to start executing and kills that worker process. It selects and kills one process at a time and waits for it to be killed before choosing another one, regardless of how frequent the monitor runs.
+
+.. dropdown:: Example: memory monitor prefers to kill a retriable task
+
+    Let's first start ray and specify the memory threshold.
+
+    .. code-block:: bash
+
+        RAY_memory_monitor_interval_ms=100 RAY_memory_usage_threshold_fraction=0.4 RAY_min_memory_free_bytes=-1 ray start --head
+
+
+    Let's create an application two_actors.py that submits two actors, where the first one is retriable and the second one is non-retriable.
+
+    .. literalinclude:: ../doc_code/ray_oom_prevention.py
+          :language: python
+          :start-after: __two_actors_start__
+          :end-before: __two_actors_end__
+
+
+    Run the application to see that only the first actor was killed.
+
+    .. code-block:: bash
+
+        $ python two_actors.py
+        
+        first actor was killed by memory monitor
+        finished second actor
+
+
+Addressing memory issues
+------------------------
+
+When the application fails due to OOM, consider increasing the memory capacity of the node or :ref:`limit the number of concurrently running tasks <core-patterns-limit-running-tasks>`.
```

Comment:
Can you also mention to inspect individual actor/task memory usage and reduce it?